storage-client/src/lib.rs

```diff
@@ -32,6 +33,9 @@ pub enum StorageRequest {
 /// underlying storage.
 #[async_trait::async_trait]
 pub trait StorageClient: Send + Sync + 'static {
+    /// Returns the physical location of the requested data in RequestParams.
+    async fn get_info_from_catalog(&self, request: StorageRequest) -> Result<RequestParams>;
+
```

Comment:
I don't think we need this method in the `StorageClient` trait. The trait should include the interfaces that the execution engine will call. EE doesn't need to send request to the catalog node. It's the storage client's responsibility. So better write this function in `impl StorageClientImpl` instead of in `impl StorageClient for StorageClientImpl`.